### PR TITLE
Support custom cell executions

### PR DESCRIPTION
**Pull request by mxsdev**
Allows running cells with custom languages. Provides new `program` cell annotation, with which the cell will run.

Requires runme with stateful/runme#302

Depends on #590

### CODE REVIEW

File: tests/extension/messages/cellOutput.test.ts [hunk 1]
```diff
@@ -78,4 +80,4 @@ suite('Handle CellOutput messages', () => {
     test('Vercel cell output', async () => {
         return expectOutput(OutputType.vercel)
     })
-})
```
Comment (duvanmonsa) [inline tests/extension/messages/cellOutput.test.ts:81]: what is the standard for us having an empty line at the end?

**Review by duvanmonsa — APPROVED** [review]
LGTM

File: tests/extension/messages/cellOutput.test.ts (hunk 1 above)
Comment (mxsdev) [reply tests/extension/messages/cellOutput.test.ts:81]: #595

File: examples/README.md [hunk 2]
```diff
@@ -151,3 +155,11 @@ echo Rows: \$(tput lines)
 echo Columns: \$(tput cols)
 "
 ```
+
+## YAML
+
+```yaml
+config:
+  netsed:
```
Comment (sourishkrout) [inline examples/README.md:163]:
```suggestion
  nested:
```

**Review by sourishkrout — CHANGES_REQUESTED** [review]
Let's talk this thru today. I'd like us to provide more clarity that this, in fact, works like a `#!` shebang. We can decide to do that in a follow-up PR (likely more UX than logic).

Outside of that something like `/usr/bin/env python3` does not work and errors will be "swallowed". Can we improve that please. Using `env` as a way to allow multi argument shebangs is commonplace.

**Comment by mxsdev** [comment]
> Let's talk this thru today. I'd like us to provide more clarity that this, in fact, works like a #! shebang. We can decide to do that in a follow-up PR (likely more UX than logic).

Agreed. I used "`program`" as a placeholder (it fits with the internal naming structure), but I presumed we would be working it out more.

> Outside of that something like /usr/bin/env python3 does not work and errors will be "swallowed". Can we improve that please. Using env as a way to allow multi argument shebangs is commonplace.

This is a bit complicated since we need to distinguish the program path from the arguments (this is probably a limitation of the syscall responsible for spawning a new process). Some paths can contain spaces, like "C:\Program Files\...". For now I can use a simple regexp on whitespace.

But eventually we would need to (in descending order of difficulty):

1. Do some basic shell parsing (maybe a use case for shlex)
2. Wrap program execution in a bash shell (makes determining if the program exists [to create error window indicating unsupported language] difficult)
3. Regexp/split on whitespace (breaks filepaths with spaces)
4. Provide a separate input field for arguments (if i recall correctly, arguments are parsed by the system/shell program, so i don't think we need to worry here)

I think I can get this working with (1), maybe in a follow-up PR.

**Comment by mxsdev** [comment]
@sourishkrout Just pushed changes that supports arguments in the `program`. One pretty massive issue, though: annotations don't allow spaces :/ not sure what to do about that.

I also fixed error handling. You should now get the following error popup if you don't provide a `program`, and the language is not supported:

```
Unable to automatically execute cell. To execute this cell, set the "program" field in the configuration foldout!
```

And the following error popup if the user-provided `program` does not exist:

```
Unable to locate program "<program>"
```

File: src/client/components/configuration/annotations.ts [hunk 3]
```diff
@@ -145,6 +145,11 @@ export class Annotations extends LitElement {
       description: "Cell's canonical name for easy referencing in the CLI.",
       docs: 'https://docs.runme.dev/configuration#cell-options',
     },
+    interpreter: {
+      description: 'Script shebang line',
+      // FIXME: update docs link
```
Comment (sourishkrout) [inline src/client/components/configuration/annotations.ts:150]: I'll do that before releasing 👍

File: src/extension/runner.ts [hunk 4]
```diff
@@ -407,6 +412,17 @@ export class GrpcRunnerProgramSession implements IRunnerProgramSession {
 
     this.session.responses.onError((error) => {
       if (error instanceof RpcError) {
+        if (error.message.includes('invalid LanguageId')) {
+          window.showErrorMessage(
+            // eslint-disable-next-line max-len
+            'Unable to automatically execute cell. To execute this cell, set the "program" field in the configuration foldout!'
```
Comment (sourishkrout) [inline src/extension/runner.ts:418]: `"interpreter" field,` let's change that prior to merging please

**Review by sourishkrout — APPROVED** [review]
👍 LGTM, just some minor things. Feel free to change & merge.

**Pull request merged**